pkg/deploy/controller/deployment/controller.go

```diff
@@ -116,16 +109,12 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		deploymentPod, err := c.podClient.createPod(deployment.Namespace, podTemplate)
 		// Retry on error.
 		if err != nil {
-			if config, decodeErr := c.decodeConfig(deployment); decodeErr == nil {
-				c.recorder.Eventf(config, kapi.EventTypeWarning, "FailedCreate", "Error creating deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err)
-			} else {
-				c.recorder.Eventf(deployment, kapi.EventTypeWarning, "FailedCreate", "Error creating deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err)
-			}
+			c.emitDeploymentEvent(deployment, config, kapi.EventTypeWarning, "FailedCreate", fmt.Sprintf("Error creating deployer pod", err))
```

Comment:
Not sure what you mean.

An event of the form "you got a conflict" is not something we should ever be sending, unless we are completely giving up.  We should only emit events when the failure is meaningful to an end user, or because of the conflict, we will not retry for a very long time.  Neither of those apply here, and given the potential for contention, we are generating meaningless events for end users.  Those events reduce comprehension and debuggability, not improve it.

So in general, the only conflict error that should result in an event being sent is when we drop an item from the retry queue until the next sync interval due to too many conflicts.